### ISSUE TITLE
Undefined offset: 10 - Matomo 4.14.2

### ISSUE THREAD

**Issue by duchengyao** [ip2location/ip2proxy-matomo#39]
`WARNING: /plugins/IP2Proxy/lib/IP2Proxy.php(1549): Notice - Undefined offset: 10 - Matomo 4.14.2 - Please report this message in the Matomo forums: https://forum.matomo.org (please do a search first as it might have been reported already) (Module: IP2Proxy, Action: getProxyDetails, Method: IP2Proxy.getProxyDetails, In CLI mode: false)`

xx.xx.xx.xx | No | - | 4ShellsAddict Proxies AnchorFreeAstrill VPNAvas | - | - | - | - | - | - | - | 2023-04-19 07:35:49 | returning | 2 | 20 分 46 秒 | direct | - | 桌面
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

<br class="Apple-interchange-newline">

**Comment by ip2location-com** [comment]
It looks like you are using the wrong BIN database. Can you provide the file name and file size for investigation?

**Comment by duchengyao** [comment]
System => general settings => ip2proxy => IP2Proxy BIN Database : `/var/www/html/matomo/misc/IP2PROXY-LITE-PX11.BIN`

`-rwxr-xr-x  1 www-data www-data 534501926 4  7 00:00  IP2PROXY-LITE-PX11.BIN*`

```
 md5sum /var/www/html/matomo/misc/IP2PROXY-LITE-PX11.BIN
5422a6720f4a2a44c299df080e84fd9f  /var/www/html/matomo/misc/IP2PROXY-LITE-PX11.BIN
```

**Comment by ip2location-com** [comment]
I'm not able to reproduce the issue from my side. Can I know the instructions to reproduce the error and the IP address of the visitor?

**Comment by duchengyao** [comment]
@ip2location 

I deployed Matomo using Git. 
All ip address are unavailable.
I'll provide any information to fix this issue if needed.

```
s1nh@s1nh-home:/var/www/html/matomo$ git show
commit 7da5cc0c383f3a1727125d3270ef12a03293dc54 (HEAD -> 4.x-dev, tag: 4.14.2, origin/next_release, origin/4.x-dev)
Author: Stefan Giehl <stefan@matomo.org>
Date:   Tue Apr 18 10:59:58 2023 +0200

    Release 4.14.2

diff --git a/core/Version.php b/core/Version.php
index 0d185fffcc..8990d88323 100644
--- a/core/Version.php
+++ b/core/Version.php
@@ -21,7 +21,7 @@ final class Version
      * The current Matomo version.
      * @var string
      */
-    const VERSION = '4.14.2-b3';
+    const VERSION = '4.14.2';
 
     const MAJOR_VERSION = 4;
```


![image](https://user-images.githubusercontent.com/7941237/234165064-015279db-11a6-4e6f-8580-223e76caf4a9.png)


```
180.169.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 02:56:54	new	1	0秒	direct	-	桌面
36.112.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 02:56:01	new	1	0秒	search	Baidu	桌面
111.205.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 02:43:25	returning	2	0秒	search	Bing	桌面
223.71.1xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 02:25:18	new	1	0秒	search	Baidu	桌面
116.30.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 02:12:13	new	1	0秒	search	Baidu	桌面
61.220.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 01:50:35	new	1	0秒	search	Google	桌面
219.150.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 01:33:07	new	1	0秒	search	Baidu	智能手机
47.94.38xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 01:15:19	new	1	0秒	direct	-	桌面
223.104.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 01:03:56	new	1	0秒	search	Google	智能手机
47.199.6xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-25 00:36:44	new	1	0秒	search	Google	桌面
218.2.10xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 16:45:10	returning	5	0秒	direct	-	桌面
120.229.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 16:40:59	new	1	0秒	search	Bing	平板电脑
101.67.4xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 16:15:26	new	1	0秒	direct	-	智能手机
157.201.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 15:44:16	new	1	0秒	search	Google	桌面
183.42.4xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 15:33:33	new	1	0秒	direct	-	智能手机
36.27.27xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 15:27:03	new	1	0秒	direct	-	桌面
66.175.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 15:26:37	new	1	0秒	direct	-	桌面
119.136.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 14:56:17	new	1	0秒	direct	-	智能手机
143.198.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 14:31:26	new	1	0秒	search	Baidu	桌面
219.140.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 14:09:08	new	1	0秒	search	Baidu	桌面
218.249.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 13:29:51	new	1	4 分 2 秒	search	Baidu	桌面
112.122.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 13:27:54	new	1	0秒	direct	-	桌面
36.113.9xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 13:25:33	returning	3	3 分 43 秒	direct	-	桌面
223.104.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 13:21:50	returning	3	0秒	search	Bing	智能手机
39.171.1xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 12:11:39	new	1	0秒	search	Baidu	桌面
125.80.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 12:11:09	new	1	0秒	direct	-	桌面
60.188.9xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 11:38:40	new	1	0秒	direct	-	智能手机
183.131.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 10:56:49	new	1	0秒	search	Baidu	桌面
27.115.1xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 10:49:17	new	1	0秒	search	Baidu	智能手机
124.236.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 10:49:08	new	1	1 分 29 秒	search	Bing	智能手机
36.32.2.xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 10:45:51	new	1	0秒	direct	-	桌面
122.193.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 10:33:48	new	1	1 分 8 秒	search	Baidu	桌面
61.150.1xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 09:43:15	new	1	0秒	search	Baidu	桌面
117.144.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 09:00:13	new	1	28 分 58 秒	search	Google	桌面
183.131.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 08:51:44	new	1	0秒	search	Baidu	桌面
39.129.5xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 08:30:06	new	1	0秒	search	Baidu	桌面
36.23.25xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 08:25:50	returning	3	20秒	direct	-	智能手机
106.38.5xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 08:14:29	new	1	0秒	search	Baidu	桌面
61.164.4xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 08:02:35	returning	2	0秒	direct	-	桌面
36.112.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:57:48	new	1	0秒	search	Bing	桌面
180.101.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:53:11	returning	4	29秒	direct	-	桌面
223.104.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:33:21	returning	2	0秒	search	Bing	智能手机
58.247.1xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:25:27	new	1	10 分 58 秒	direct	-	桌面
113.116.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:25:19	new	1	4 分 45 秒	direct	-	智能手机
61.164.4xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:20:27	returning	3	47秒	direct	-	桌面
180.98.1xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:18:08	new	1	0秒	search	Baidu	桌面
39.163.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:17:20	new	1	0秒	search	Baidu	智能手机
36.98.24xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:14:50	new	1	0秒	search	Baidu	平板手机
61.172.2xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 07:02:10	new	1	4 分 32 秒	search	Baidu	桌面
124.116.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 06:49:09	new	1	0秒	search	Baidu	桌面
106.39.0xx.xx	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 06:48:47	new	1	0秒	direct	-	桌面
117.131.xx.xx	No	-	4ShellsAddict Proxies AnchorFreeAstrill VPNAvas	-	-	-	-	-	-	-	2023-04-24 06:37:12	new	1	0秒	search	Google	桌面
```

**Comment by duchengyao** [comment]
@ip2location is there any way to export logs of this plugin?

**Comment by ip2location-com** [comment]
Can you confirm that you have uploaded the BIN file in binary mode to the server?

It looks like there are some ASCII codes in the result.

**Comment by duchengyao** [comment]
@ip2location 

Yes, I uploaded the bin file using ssh.

I have posted md5sum above.

**Comment by ip2location-com** [comment]
The issue is resolved in version 0.1.5.
